fluxv2.tf

```diff
@@ -0,0 +1,195 @@
+data "kubectl_file_documents" "sync" {
+  count   = local.fluxv2["enabled"] ? 1 : 0
+  content = data.flux_sync.main[0].content
+}
+
+# Apply manifests on the cluster
+resource "kubectl_manifest" "sync" {
+  for_each   = { for v in local.sync : lower(join("/", compact([v.data.apiVersion, v.data.kind, lookup(v.data.metadata, "namespace", ""), v.data.metadata.name]))) => v.content }
+  depends_on = [kubernetes_namespace.fluxv2]
+  yaml_body  = each.value
+}
+
+# Generate a Kubernetes secret with the Git credentials
+resource "kubernetes_secret" "main" {
+  count      = local.fluxv2["enabled"] ? 1 : 0
+  depends_on = [kubectl_manifest.apply]
+
+  metadata {
+    name      = data.flux_sync.main[0].name
+    namespace = data.flux_sync.main[0].namespace
+  }
+
+  data = {
+    "identity.pub" = tls_private_key.identity[0].public_key_pem
+    identity       = tls_private_key.identity[0].private_key_pem
+    username       = "git"
+    password       = local.fluxv2["github_token"]
+  }
+}
+
+# GitHub
+resource "github_repository" "main" {
+  count      = local.fluxv2["enabled"] ? 1 : 0
+  name       = local.fluxv2["repository"]
+  visibility = local.fluxv2["repository_visibility"]
+  auto_init  = true
+}
+
+resource "github_branch_default" "main" {
+  count      = local.fluxv2["enabled"] ? 1 : 0
+  repository = github_repository.main[0].name
+  branch     = local.fluxv2["branch"]
+}
+
+resource "github_repository_deploy_key" "main" {
+  count      = local.fluxv2["enabled"] ? 1 : 0
+  title      = "flux-${github_repository.main[0].name}-${local.fluxv2["branch"]}"
+  repository = github_repository.main[0].name
+  key        = tls_private_key.identity[0].public_key_openssh
+  read_only  = true
+}
+
+resource "github_repository_file" "install" {
+  count      = local.fluxv2["enabled"] ? 1 : 0
+  repository = github_repository.main[0].name
+  file       = data.flux_install.main[0].path
+  content    = data.flux_install.main[0].content
+  branch     = local.fluxv2["branch"]
+}
+
+resource "github_repository_file" "sync" {
+  count      = local.fluxv2["enabled"] ? 1 : 0
+  repository = github_repository.main[0].name
+  file       = data.flux_sync.main[0].path
+  content    = data.flux_sync.main[0].content
+  branch     = local.fluxv2["branch"]
+}
+
+resource "github_repository_file" "kustomize" {
+  count      = local.fluxv2["enabled"] ? 1 : 0
+  repository = github_repository.main[0].name
+  file       = data.flux_sync.main[0].kustomize_path
+  content    = data.flux_sync.main[0].kustomize_content
+  branch     = local.fluxv2["branch"]
+}
+
+resource "kubernetes_network_policy" "fluxv2_allow_monitoring" {
+  count = local.fluxv2["enabled"] && local.fluxv2["network_policy"] && local.kube-prometheus-stack["enabled"] ? 1 : 0
+
+  metadata {
+    name      = "${kubernetes_namespace.fluxv2.*.metadata.0.name[count.index]}-allow-monitoring"
+    namespace = kubernetes_namespace.fluxv2.*.metadata.0.name[count.index]
```

Comment:
```
    name      = "${local.fluxv2["create_ns"] ? kubernetes_namespace.fluxv2.*.metadata.0.name[count.index] : local.fluxv2["namespace"]}-allow-ingress-fluxv2"
    namespace = local.fluxv2["create_ns"] ? kubernetes_namespace.fluxv2.*.metadata.0.name[count.index] : local.fluxv2["namespace"]
```

Reply:
Whether `create_ns` is used or not, the name and the namespace will always be the same so we could just use `local.flux2["namespace]` as metadata name and metadata namespace.

Reply:
yes but if you create the namespace this create an implicit dependency so terraform does not try to deploy the chart before the NS, it is needed when you create the namespace